core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -175,14 +177,13 @@ private AggregationAnalyzer(
                 .map(expression -> scopeAwareKey(expression, analysis, sourceScope))
                 .collect(toImmutableSet());
 
-        this.columnReferences = analysis.getColumnReferenceFields()
-                .entrySet().stream()
-                .collect(toImmutableMap(Map.Entry::getKey, entry -> entry.getValue().getFieldId()));
+        this.columnReferences = analysis.getColumnReferenceFields();
 
         this.groupingFields = groupByExpressions.stream()
                 .map(NodeRef::of)
                 .filter(columnReferences::containsKey)
                 .map(columnReferences::get)
+                .map(ResolvedField::getFieldId)
```

Comment:
Would it help perf if we joined

```
.filter(columnReferences::containsKey)
                .map(columnReferences::get)
```

into one map call?

for example, in similar code below we do

```
.map(columnReferences::get)
                                    .filter(Objects::nonNull)
```

Reply:
As the hugely overrated and usually incorrectly quoted quotation says
"premature optimization is the root of all evil".
This is not a bottleneck. The stream iterates over grouping fields which by no means reach any significant number.
If this was a bottleneck using a stream itself would be the biggest problem.